static/app/components/dropdownMenuControl.tsx

```diff
@@ -163,7 +163,7 @@ function DropdownMenuControl({
       triggerRef.current !== target &&
       !triggerRef.current?.contains(target),
     // Necessary for submenus to be correctly positioned
-    ...(isSubmenu && {preventOverflowOptions: {boundary: document.body}}),
+    ...(isSubmenu && {preventOverflowOptions: {boundary: document.body, altAxis: true}}),
```

Comment:
Tell PopperJS to prevent overflow on the alt axis (which in this case is the horizontal axis)